Improve Recommendations for Form Validation

yih-wang opened the pull request:
- Add rules of when to perform validation and when to disable submit button.
- Add a subtitle to the first paragraph as “How to perform validation”.

This pr links to the issue #388

mcarrano commented on pattern-library/forms-and-controls/errors-and-validation/design/overview.md:
```diff
@@ -2,14 +2,23 @@
 
 ![Error and Validation](img/error-and-validation.png)
 
+### When to perform validation
```
I think that all of these examples are instances of client-side validation. Should we just explain here that there are two options for client-side or server-side validation and explain what each is? This would consolidate this a bit and feel less redundant.

mcarrano commented on pattern-library/forms-and-controls/errors-and-validation/design/overview.md:
```diff
@@ -2,14 +2,23 @@
-* Note: When a form is submitted, progress indicators (spinner or progress bars) should be used if the time to process the form submit action would exceed users’ expectations. For more information about progress indicators, see: [spinner](https://www.patternfly.org/widgets/#spinner) and/or [progress](https://www.patternfly.org/widgets/#progress).
+* An (optional) inline notification might be presented as well, depending on the use case. See “Server-side validation”.
 
 ![Error and Validation](img/error-and-validation.png)
```
While we are making these changes, can we also update this mockup per #436 so that the button labels are correct?

yih-wang commented:
@mcarrano I think it's better to introduce the two types of validation: client-side validation and server-side validation, then list how and when to perform the validations. So I restructured the guidelines, adjusted the order of paragraphs. Also remove some similar expressions. As a big change was made, I updated the guidelines in the Google Doc for this step, you can review it from here: https://docs.google.com/a/redhat.com/document/d/1lLxifpKba1siMg8OEGW_bLByUYmshygffZBE0pGPXzI/edit?usp=sharing

yih-wang commented:
@mcarrano Modified the guidelines according to the draft we finished in Google Doc. Both [Errors and Validation](http://www.patternfly.org/pattern-library/forms-and-controls/errors-and-validation/) and [Buttons on Forms](http://www.patternfly.org/pattern-library/forms-and-controls/buttons-on-forms/) pages have been updated.

mcarrano commented on pattern-library/forms-and-controls/buttons-on-forms/design/overview.md:
```diff
@@ -34,9 +34,10 @@ Buttons may also be used for inline actions within a form.  In these cases, the
 
 ### Enabling and Disabling Actions
-Submit buttons should be disabled when all conditions required for completing a form are not met.  This will reduce the need to return validation errors because the user neglected to complete the form.  When using disabled buttons, it should always be possible for the user to enable the button by inputing information onto the current page.  Do not use this technique for blocking functionality tha ...
+* For one-page experience, submit buttons should be disabled when all conditions required for completing a form are not met. This will reduce the need to return validation errors because the user neglected to complete the form. However, in the cases of long scrolling-page forms, we recommend against do like that as the submit button is not always visible. When using disabled buttons, it should al ...
```
Can we change to, "For short forms (e.g. wizards or similar forms), the submit buttons should be disabled until all conditions required for completing a form are met. This will reduce the need to return validation errors because the user neglected to complete the form. However, in cases of long scrolling-page forms, we recommend against disabling the submit button as the missing field and the button may not be visible on the screen at the same time.

mcarrano commented:
@catrobson @serenamarie125 @maryclarke @jeperry Can you take a look and make sure we are all in agreement on these recommendations.

yih-wang commented:
@mcarrano Thanks for the help of a better wording. I modified the corresponding paragraph and commit a new version.
